scipy/stats/_stats_py.py

```diff
@@ -9440,3 +9442,88 @@ def rankdata(a, method='average', *, axis=None, nan_policy='propagate'):
         result[nan_indexes] = np.nan
 
     return result
+
+
+# TODO: axis parameter
+# TODO: add _axis_nan_policy_factory decorator
+def expectile(a, alpha=0.5, axis=0, dtype=None, weights=None):
+    r"""Compute the expectile along the specified axis.
+
+    The expectile at level alpha of the array :math:`a_i` associated to
+    weights :math:`w_i` is the unique solution :math:`t` of:
+
+    .. math::
+
+        \alpha \sum_{i=1}^n w_i (y_i - t)_+ =
+            (1 - \alpha) \sum_{i=1}^n w_i (t - y_i)_+ \,.
+
+    Expectiles are a generalization of the expectation in the same way as
+    quantiles are a generalization of the median. The expectile at level
+    `alpha = 0.5` is the mean (average).
+
+    Parameters
+    ----------
+    a : array_like
+        Input array or object that can be converted to an array.
+    alpha : float
+        The level of the expectile; `alpha=0.5` give the mean.
+    axis : int or None, optional
+        Axis along which the geometric mean is computed. Default is 0.
+        If None, compute over the whole array `a`.
+    dtype : dtype, optional
+        Type to which the input arrays are cast before the calculation is
+        performed.
+    weights : array_like, optional
+        The `weights` array must be broadcastable to the same shape as `a`.
+        Default is None, which gives each value a weight of 1.0.
+
+    Returns
+    -------
+    expectile : ndarray
+        See `dtype` parameter above.
+
+    See Also
+    --------
+    numpy.mean : Arithmetic average
+    numpy.quantile : Quantile
+
+    References
+    ----------
+    .. [1] W. K. Newey and J. L. Powell (1987), "Asymmetric Least Squares
+           Estimation and Testing," Econometrica, 55, 819-847.
+
+    Examples
+    --------
+    >>> from scipy.stats import gmean
```

Comment:
Please show some of the properties with the examples (e.g. compare against mean).